object_storage_api/services/image.py

```diff
@@ -85,3 +85,13 @@ def list(self, entity_id: Optional[str] = None, primary: Optional[bool] = None)
         """
         images = self._image_repository.list(entity_id, primary)
         return [ImageSchema(**image.model_dump()) for image in images]
+
+    def delete(self, image_id: str) -> None:
+        """
+        Delete an image by its ID.
+
+        :param image_id: The ID of the image to delete.
+        """
+
+        object_key = self._image_repository.delete(image_id)
+        self._image_store.delete(object_key)
```

Comment:
Its just because these tasks have ` Must gracefully deal with a situation when the file object with the given object_key doesn't exist` in them so if its deleted first and it ends up not having the file then that is considered ok. But we dont want the other way round as the way we would look up things in the object storage is using the database.

Also the transaction wouldn't work the other way round as its a MongoDB thing - in this case it would ensures that if an error occurs before it is finished, none of the changes actual end up committed to the database, so in this case if it was wrapped in a transaction if self._image_store.delete raised any error the first database edit would never be finalised.